docs/ui-guidelines.md

```diff
@@ -0,0 +1,36 @@
+The vast majority of the UI, including CLI plugins, is to be built using our CLI design system. This ensures a familiar-feeling interction patterns and reduces the complexity behind-the-scenes. It also helps our teams work faster and rely on validation rules
```

Comment:
@MeredithCastile what about moving this to `docs/cli-kit` and adding a reference to [this page](https://github.com/Shopify/cli/tree/main/docs)